bundles/org.smarthomej.transform.chain/src/main/java/org/smarthomej/transform/chain/internal/ChainTransformationProfileFactory.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2021 Contributors to the SmartHome/J project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.smarthomej.transform.chain.internal;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.thing.profiles.Profile;
+import org.openhab.core.thing.profiles.ProfileCallback;
+import org.openhab.core.thing.profiles.ProfileContext;
+import org.openhab.core.thing.profiles.ProfileFactory;
+import org.openhab.core.thing.profiles.ProfileType;
+import org.openhab.core.thing.profiles.ProfileTypeBuilder;
+import org.openhab.core.thing.profiles.ProfileTypeProvider;
+import org.openhab.core.thing.profiles.ProfileTypeUID;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.smarthomej.commons.transform.ValueTransformationProvider;
+
+/**
+ *
+ * A Factory that creates the transformation profiles for the {@link ChainTransformationProfile}
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = { ProfileFactory.class, ProfileTypeProvider.class })
+public class ChainTransformationProfileFactory implements ProfileFactory, ProfileTypeProvider {
+    private final ValueTransformationProvider valueTransformationProvider;
+
+    @Activate
+    public ChainTransformationProfileFactory(@Reference ValueTransformationProvider valueTransformationProvider) {
+        this.valueTransformationProvider = valueTransformationProvider;
+    }
+
+    @Override
+    public Collection<ProfileType> getProfileTypes(@Nullable Locale locale) {
+        return List.of(ProfileTypeBuilder.newState(ChainTransformationProfile.PROFILE_TYPE_UID,
+                ChainTransformationProfile.PROFILE_TYPE_UID.getId()).build());
+    }
+
+    @Override
+    public @Nullable Profile createProfile(ProfileTypeUID profileTypeUID, ProfileCallback callback,
+            ProfileContext profileContext) {
+        return new ChainTransformationProfile(callback, profileContext, valueTransformationProvider);
```

Comment:
IIRC you should check for matching `ProfileTypeUID` or return `null` otherwise. I am not sure if the calling instance checks the factory for `getSupportedProfileTypeUIDs()` or not.

```suggestion
        return ChainTransformationProfile.PROFILE_TYPE_UID.equals(profileTypeUID) ? new ChainTransformationProfile(callback, profileContext, valueTransformationProvider) : null;
```